test/controllers/authorizeService.js

```diff
@@ -0,0 +1,184 @@
+var proxyquire = require('proxyquire').noCallThru();
+
+describe('Accounting Service', function () {
+
+     var DEFAULT_URL = 'http://example/path';
+     var DEFAULT_WSTOREHOST = 'localhost';
+     var DEFAULT_APIKEY = 'apiKey';
+
+     var getAuthorizeServiceController = function (accServiceSchema, config) {
+          return proxyquire('../../controllers/authorizeService', {
+               '../db/schemas/accountingService': accServiceSchema,
+               '../config': config
+          }).authorizeService;
+     };
+
+     var invalidRequest = function (handler, req, statusExpected, bodyExpected, done) {
+          var config = {
+               appHost: DEFAULT_WSTOREHOST
+          };
+
+          var accSerivceController = getAuthorizeServiceController({}, config);
+
+          res = jasmine.createSpyObj('res', ['status', 'json']);
+
+          res.status.and.returnValue(res);
+
+          accSerivceController[handler](req, res);
+
+          setTimeout(function () {
+
+               expect(res.status).toHaveBeenCalledWith(statusExpected);
+               expect(res.json).toHaveBeenCalledWith(bodyExpected);
+
+               done();
+
+          }, 100);
+     };
+
+     describe('Get api-key', function () {
+
+          it('should return 401 when the requester is not the WStore', function (done) {
+               var req = {
+                    ip: {
+                         replace: function (expr) { return '130.25.13.12'}
+                    }
+               };
+
+               invalidRequest('getApiKey', req, 401, {error: 'Invalid remote client'}, done);
+          });
+
+          it('should return 400 when the body is empty', function (done) {
+               var req = {
```

Comment:
This code is repeated. Just create `req` in the `invalidRequest` function and pass to the function the IP that should be generated by the `replace` function. Use a mock to generate the request.